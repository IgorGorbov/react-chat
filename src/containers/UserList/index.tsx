import * as React from 'react';
import UserCard from '../../components/UserCard';
import { UserListWrapper } from '../../styledComponents';

class UserList extends React.Component {
  public static renderUserCard(user: IUserCard): JSX.Element {
    return <UserCard key={user.id} {...user} />;
  }

  public render() {
    const users: IUserCard[] = [
      {
        id: 2,
        name: 'Louis Litt',
        avatar: 'https://emilcarlsson.se/assets/louislitt.png',
      },
      {
        id: 3,
        name: 'Louis Litt',
        avatar: 'https://emilcarlsson.se/assets/louislitt.png',
      },
      {
        id: 4,
        name: 'Louis Litt',
        avatar: 'https://emilcarlsson.se/assets/louislitt.png',
      },
      {
        id: 5,
        name: 'Louis Litt',
        avatar: 'https://emilcarlsson.se/assets/louislitt.png',
      },
      {
        id: 6,
        name: 'Louis Litt',
        avatar: 'https://emilcarlsson.se/assets/louislitt.png',
      },
      {
        id: 7,
        name: 'Louis Litt',
        avatar: 'https://emilcarlsson.se/assets/louislitt.png',
      },
      {
        id: 8,
        name: 'Louis Litt',
        avatar: 'https://emilcarlsson.se/assets/louislitt.png',
      },
      {
        id: 9,
        name: 'Louis Litt',
        avatar: 'https://emilcarlsson.se/assets/louislitt.png',
      },
      {
        id: 10,
        name: 'Louis Litt',
        avatar: 'https://emilcarlsson.se/assets/louislitt.png',
      },
    ];

    return (
      <UserListWrapper>{users.map(UserList.renderUserCard)}</UserListWrapper>
    );
  }
}

export default UserList;
