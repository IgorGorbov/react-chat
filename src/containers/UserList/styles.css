.UserListWrapper {
    max-height: 480px;
    overflow-y: scroll;
}
