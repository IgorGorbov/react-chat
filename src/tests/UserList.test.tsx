import * as React from "react";
import toJSON from 'enzyme-to-json';
import { mount } from "enzyme";
import UserList from "../containers/UserList";
import {
  UserListWrapper
} from '../StyledComponents'

describe('testing UserList', () => {
  const wrapper = mount(<UserList />);

  it("should contain UserListWrapper", () => {
    expect(wrapper.find(UserListWrapper)).toHaveLength(1);
  });

  it('snapshot UserList', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  })
});
