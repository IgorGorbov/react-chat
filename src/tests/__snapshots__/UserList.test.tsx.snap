// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UserList snapshot UserList 1`] = `
<UserList>
  <styled.div>
    <div
      className="sc-fjdhpX lcnmRg"
    >
      <UserCard
        avatar="https://emilcarlsson.se/assets/louislitt.png"
        id={2}
        key="2"
        name="Louis Litt"
      >
        <styled.div>
          <div
            className="sc-jzJRlG ikoaaf"
          >
            <styled.div>
              <div
                className="sc-cSHVUG hDUXUW"
              >
                <Styled(Image)
                  bordercolor="white"
                  bordersize="1px"
                  circular={true}
                  maxheight="65px"
                  maxwidth="65px"
                  size="tiny"
                  src="https://emilcarlsson.se/assets/louislitt.png"
                >
                  <Image
                    as="img"
                    bordercolor="white"
                    bordersize="1px"
                    circular={true}
                    className="sc-htpNat drfAkD"
                    maxheight="65px"
                    maxwidth="65px"
                    size="tiny"
                    src="https://emilcarlsson.se/assets/louislitt.png"
                    ui={true}
                  >
                    <img
                      bordercolor="white"
                      bordersize="1px"
                      className="ui tiny circular image sc-htpNat drfAkD"
                      maxheight="65px"
                      maxwidth="65px"
                      src="https://emilcarlsson.se/assets/louislitt.png"
                    />
                  </Image>
                </Styled(Image)>
                <styled.span>
                  <span
                    className="sc-kAzzGY bhplhq"
                  />
                </styled.span>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-chPdSV jmNmdm"
              >
                <styled.p>
                  <p
                    className="sc-kgoBCf eyQkMk"
                  >
                    Louis Litt
                  </p>
                </styled.p>
                <styled.p>
                  <p
                    className="sc-kGXeez bpwcQb"
                  >
                    Lorem ipsum dolor sit amet.
                  </p>
                </styled.p>
              </div>
            </styled.div>
            <styled.p>
              <p
                className="sc-kpOJdX fuYgIr"
              >
                3
              </p>
            </styled.p>
          </div>
        </styled.div>
      </UserCard>
      <UserCard
        avatar="https://emilcarlsson.se/assets/louislitt.png"
        id={3}
        key="3"
        name="Louis Litt"
      >
        <styled.div>
          <div
            className="sc-jzJRlG ikoaaf"
          >
            <styled.div>
              <div
                className="sc-cSHVUG hDUXUW"
              >
                <Styled(Image)
                  bordercolor="white"
                  bordersize="1px"
                  circular={true}
                  maxheight="65px"
                  maxwidth="65px"
                  size="tiny"
                  src="https://emilcarlsson.se/assets/louislitt.png"
                >
                  <Image
                    as="img"
                    bordercolor="white"
                    bordersize="1px"
                    circular={true}
                    className="sc-htpNat drfAkD"
                    maxheight="65px"
                    maxwidth="65px"
                    size="tiny"
                    src="https://emilcarlsson.se/assets/louislitt.png"
                    ui={true}
                  >
                    <img
                      bordercolor="white"
                      bordersize="1px"
                      className="ui tiny circular image sc-htpNat drfAkD"
                      maxheight="65px"
                      maxwidth="65px"
                      src="https://emilcarlsson.se/assets/louislitt.png"
                    />
                  </Image>
                </Styled(Image)>
                <styled.span>
                  <span
                    className="sc-kAzzGY bhplhq"
                  />
                </styled.span>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-chPdSV jmNmdm"
              >
                <styled.p>
                  <p
                    className="sc-kgoBCf eyQkMk"
                  >
                    Louis Litt
                  </p>
                </styled.p>
                <styled.p>
                  <p
                    className="sc-kGXeez bpwcQb"
                  >
                    Lorem ipsum dolor sit amet.
                  </p>
                </styled.p>
              </div>
            </styled.div>
            <styled.p>
              <p
                className="sc-kpOJdX fuYgIr"
              >
                3
              </p>
            </styled.p>
          </div>
        </styled.div>
      </UserCard>
      <UserCard
        avatar="https://emilcarlsson.se/assets/louislitt.png"
        id={4}
        key="4"
        name="Louis Litt"
      >
        <styled.div>
          <div
            className="sc-jzJRlG ikoaaf"
          >
            <styled.div>
              <div
                className="sc-cSHVUG hDUXUW"
              >
                <Styled(Image)
                  bordercolor="white"
                  bordersize="1px"
                  circular={true}
                  maxheight="65px"
                  maxwidth="65px"
                  size="tiny"
                  src="https://emilcarlsson.se/assets/louislitt.png"
                >
                  <Image
                    as="img"
                    bordercolor="white"
                    bordersize="1px"
                    circular={true}
                    className="sc-htpNat drfAkD"
                    maxheight="65px"
                    maxwidth="65px"
                    size="tiny"
                    src="https://emilcarlsson.se/assets/louislitt.png"
                    ui={true}
                  >
                    <img
                      bordercolor="white"
                      bordersize="1px"
                      className="ui tiny circular image sc-htpNat drfAkD"
                      maxheight="65px"
                      maxwidth="65px"
                      src="https://emilcarlsson.se/assets/louislitt.png"
                    />
                  </Image>
                </Styled(Image)>
                <styled.span>
                  <span
                    className="sc-kAzzGY bhplhq"
                  />
                </styled.span>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-chPdSV jmNmdm"
              >
                <styled.p>
                  <p
                    className="sc-kgoBCf eyQkMk"
                  >
                    Louis Litt
                  </p>
                </styled.p>
                <styled.p>
                  <p
                    className="sc-kGXeez bpwcQb"
                  >
                    Lorem ipsum dolor sit amet.
                  </p>
                </styled.p>
              </div>
            </styled.div>
            <styled.p>
              <p
                className="sc-kpOJdX fuYgIr"
              >
                3
              </p>
            </styled.p>
          </div>
        </styled.div>
      </UserCard>
      <UserCard
        avatar="https://emilcarlsson.se/assets/louislitt.png"
        id={5}
        key="5"
        name="Louis Litt"
      >
        <styled.div>
          <div
            className="sc-jzJRlG ikoaaf"
          >
            <styled.div>
              <div
                className="sc-cSHVUG hDUXUW"
              >
                <Styled(Image)
                  bordercolor="white"
                  bordersize="1px"
                  circular={true}
                  maxheight="65px"
                  maxwidth="65px"
                  size="tiny"
                  src="https://emilcarlsson.se/assets/louislitt.png"
                >
                  <Image
                    as="img"
                    bordercolor="white"
                    bordersize="1px"
                    circular={true}
                    className="sc-htpNat drfAkD"
                    maxheight="65px"
                    maxwidth="65px"
                    size="tiny"
                    src="https://emilcarlsson.se/assets/louislitt.png"
                    ui={true}
                  >
                    <img
                      bordercolor="white"
                      bordersize="1px"
                      className="ui tiny circular image sc-htpNat drfAkD"
                      maxheight="65px"
                      maxwidth="65px"
                      src="https://emilcarlsson.se/assets/louislitt.png"
                    />
                  </Image>
                </Styled(Image)>
                <styled.span>
                  <span
                    className="sc-kAzzGY bhplhq"
                  />
                </styled.span>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-chPdSV jmNmdm"
              >
                <styled.p>
                  <p
                    className="sc-kgoBCf eyQkMk"
                  >
                    Louis Litt
                  </p>
                </styled.p>
                <styled.p>
                  <p
                    className="sc-kGXeez bpwcQb"
                  >
                    Lorem ipsum dolor sit amet.
                  </p>
                </styled.p>
              </div>
            </styled.div>
            <styled.p>
              <p
                className="sc-kpOJdX fuYgIr"
              >
                3
              </p>
            </styled.p>
          </div>
        </styled.div>
      </UserCard>
      <UserCard
        avatar="https://emilcarlsson.se/assets/louislitt.png"
        id={6}
        key="6"
        name="Louis Litt"
      >
        <styled.div>
          <div
            className="sc-jzJRlG ikoaaf"
          >
            <styled.div>
              <div
                className="sc-cSHVUG hDUXUW"
              >
                <Styled(Image)
                  bordercolor="white"
                  bordersize="1px"
                  circular={true}
                  maxheight="65px"
                  maxwidth="65px"
                  size="tiny"
                  src="https://emilcarlsson.se/assets/louislitt.png"
                >
                  <Image
                    as="img"
                    bordercolor="white"
                    bordersize="1px"
                    circular={true}
                    className="sc-htpNat drfAkD"
                    maxheight="65px"
                    maxwidth="65px"
                    size="tiny"
                    src="https://emilcarlsson.se/assets/louislitt.png"
                    ui={true}
                  >
                    <img
                      bordercolor="white"
                      bordersize="1px"
                      className="ui tiny circular image sc-htpNat drfAkD"
                      maxheight="65px"
                      maxwidth="65px"
                      src="https://emilcarlsson.se/assets/louislitt.png"
                    />
                  </Image>
                </Styled(Image)>
                <styled.span>
                  <span
                    className="sc-kAzzGY bhplhq"
                  />
                </styled.span>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-chPdSV jmNmdm"
              >
                <styled.p>
                  <p
                    className="sc-kgoBCf eyQkMk"
                  >
                    Louis Litt
                  </p>
                </styled.p>
                <styled.p>
                  <p
                    className="sc-kGXeez bpwcQb"
                  >
                    Lorem ipsum dolor sit amet.
                  </p>
                </styled.p>
              </div>
            </styled.div>
            <styled.p>
              <p
                className="sc-kpOJdX fuYgIr"
              >
                3
              </p>
            </styled.p>
          </div>
        </styled.div>
      </UserCard>
      <UserCard
        avatar="https://emilcarlsson.se/assets/louislitt.png"
        id={7}
        key="7"
        name="Louis Litt"
      >
        <styled.div>
          <div
            className="sc-jzJRlG ikoaaf"
          >
            <styled.div>
              <div
                className="sc-cSHVUG hDUXUW"
              >
                <Styled(Image)
                  bordercolor="white"
                  bordersize="1px"
                  circular={true}
                  maxheight="65px"
                  maxwidth="65px"
                  size="tiny"
                  src="https://emilcarlsson.se/assets/louislitt.png"
                >
                  <Image
                    as="img"
                    bordercolor="white"
                    bordersize="1px"
                    circular={true}
                    className="sc-htpNat drfAkD"
                    maxheight="65px"
                    maxwidth="65px"
                    size="tiny"
                    src="https://emilcarlsson.se/assets/louislitt.png"
                    ui={true}
                  >
                    <img
                      bordercolor="white"
                      bordersize="1px"
                      className="ui tiny circular image sc-htpNat drfAkD"
                      maxheight="65px"
                      maxwidth="65px"
                      src="https://emilcarlsson.se/assets/louislitt.png"
                    />
                  </Image>
                </Styled(Image)>
                <styled.span>
                  <span
                    className="sc-kAzzGY bhplhq"
                  />
                </styled.span>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-chPdSV jmNmdm"
              >
                <styled.p>
                  <p
                    className="sc-kgoBCf eyQkMk"
                  >
                    Louis Litt
                  </p>
                </styled.p>
                <styled.p>
                  <p
                    className="sc-kGXeez bpwcQb"
                  >
                    Lorem ipsum dolor sit amet.
                  </p>
                </styled.p>
              </div>
            </styled.div>
            <styled.p>
              <p
                className="sc-kpOJdX fuYgIr"
              >
                3
              </p>
            </styled.p>
          </div>
        </styled.div>
      </UserCard>
      <UserCard
        avatar="https://emilcarlsson.se/assets/louislitt.png"
        id={8}
        key="8"
        name="Louis Litt"
      >
        <styled.div>
          <div
            className="sc-jzJRlG ikoaaf"
          >
            <styled.div>
              <div
                className="sc-cSHVUG hDUXUW"
              >
                <Styled(Image)
                  bordercolor="white"
                  bordersize="1px"
                  circular={true}
                  maxheight="65px"
                  maxwidth="65px"
                  size="tiny"
                  src="https://emilcarlsson.se/assets/louislitt.png"
                >
                  <Image
                    as="img"
                    bordercolor="white"
                    bordersize="1px"
                    circular={true}
                    className="sc-htpNat drfAkD"
                    maxheight="65px"
                    maxwidth="65px"
                    size="tiny"
                    src="https://emilcarlsson.se/assets/louislitt.png"
                    ui={true}
                  >
                    <img
                      bordercolor="white"
                      bordersize="1px"
                      className="ui tiny circular image sc-htpNat drfAkD"
                      maxheight="65px"
                      maxwidth="65px"
                      src="https://emilcarlsson.se/assets/louislitt.png"
                    />
                  </Image>
                </Styled(Image)>
                <styled.span>
                  <span
                    className="sc-kAzzGY bhplhq"
                  />
                </styled.span>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-chPdSV jmNmdm"
              >
                <styled.p>
                  <p
                    className="sc-kgoBCf eyQkMk"
                  >
                    Louis Litt
                  </p>
                </styled.p>
                <styled.p>
                  <p
                    className="sc-kGXeez bpwcQb"
                  >
                    Lorem ipsum dolor sit amet.
                  </p>
                </styled.p>
              </div>
            </styled.div>
            <styled.p>
              <p
                className="sc-kpOJdX fuYgIr"
              >
                3
              </p>
            </styled.p>
          </div>
        </styled.div>
      </UserCard>
      <UserCard
        avatar="https://emilcarlsson.se/assets/louislitt.png"
        id={9}
        key="9"
        name="Louis Litt"
      >
        <styled.div>
          <div
            className="sc-jzJRlG ikoaaf"
          >
            <styled.div>
              <div
                className="sc-cSHVUG hDUXUW"
              >
                <Styled(Image)
                  bordercolor="white"
                  bordersize="1px"
                  circular={true}
                  maxheight="65px"
                  maxwidth="65px"
                  size="tiny"
                  src="https://emilcarlsson.se/assets/louislitt.png"
                >
                  <Image
                    as="img"
                    bordercolor="white"
                    bordersize="1px"
                    circular={true}
                    className="sc-htpNat drfAkD"
                    maxheight="65px"
                    maxwidth="65px"
                    size="tiny"
                    src="https://emilcarlsson.se/assets/louislitt.png"
                    ui={true}
                  >
                    <img
                      bordercolor="white"
                      bordersize="1px"
                      className="ui tiny circular image sc-htpNat drfAkD"
                      maxheight="65px"
                      maxwidth="65px"
                      src="https://emilcarlsson.se/assets/louislitt.png"
                    />
                  </Image>
                </Styled(Image)>
                <styled.span>
                  <span
                    className="sc-kAzzGY bhplhq"
                  />
                </styled.span>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-chPdSV jmNmdm"
              >
                <styled.p>
                  <p
                    className="sc-kgoBCf eyQkMk"
                  >
                    Louis Litt
                  </p>
                </styled.p>
                <styled.p>
                  <p
                    className="sc-kGXeez bpwcQb"
                  >
                    Lorem ipsum dolor sit amet.
                  </p>
                </styled.p>
              </div>
            </styled.div>
            <styled.p>
              <p
                className="sc-kpOJdX fuYgIr"
              >
                3
              </p>
            </styled.p>
          </div>
        </styled.div>
      </UserCard>
      <UserCard
        avatar="https://emilcarlsson.se/assets/louislitt.png"
        id={10}
        key="10"
        name="Louis Litt"
      >
        <styled.div>
          <div
            className="sc-jzJRlG ikoaaf"
          >
            <styled.div>
              <div
                className="sc-cSHVUG hDUXUW"
              >
                <Styled(Image)
                  bordercolor="white"
                  bordersize="1px"
                  circular={true}
                  maxheight="65px"
                  maxwidth="65px"
                  size="tiny"
                  src="https://emilcarlsson.se/assets/louislitt.png"
                >
                  <Image
                    as="img"
                    bordercolor="white"
                    bordersize="1px"
                    circular={true}
                    className="sc-htpNat drfAkD"
                    maxheight="65px"
                    maxwidth="65px"
                    size="tiny"
                    src="https://emilcarlsson.se/assets/louislitt.png"
                    ui={true}
                  >
                    <img
                      bordercolor="white"
                      bordersize="1px"
                      className="ui tiny circular image sc-htpNat drfAkD"
                      maxheight="65px"
                      maxwidth="65px"
                      src="https://emilcarlsson.se/assets/louislitt.png"
                    />
                  </Image>
                </Styled(Image)>
                <styled.span>
                  <span
                    className="sc-kAzzGY bhplhq"
                  />
                </styled.span>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-chPdSV jmNmdm"
              >
                <styled.p>
                  <p
                    className="sc-kgoBCf eyQkMk"
                  >
                    Louis Litt
                  </p>
                </styled.p>
                <styled.p>
                  <p
                    className="sc-kGXeez bpwcQb"
                  >
                    Lorem ipsum dolor sit amet.
                  </p>
                </styled.p>
              </div>
            </styled.div>
            <styled.p>
              <p
                className="sc-kpOJdX fuYgIr"
              >
                3
              </p>
            </styled.p>
          </div>
        </styled.div>
      </UserCard>
    </div>
  </styled.div>
</UserList>
`;
